pkg/controller/controller.go

```
@@ -266,10 +261,21 @@ func NewController(opts ControllerOptions, deps *Dependencies) (Controller, erro
 	return nil, fmt.Errorf("no valid controller configuration provided")
 }
 
-func (c *investigationRunner) runInvestigation(ctx context.Context, clusterId string, inv investigation.Investigation, pdClient *pagerduty.SdkClient, filterCtx *types.FilterContext, params map[string]string) error {
-	metrics.Inc(metrics.Alerts, inv.Name())
+// runChain executes a config-defined chain of investigations.
+func (c *investigationRunner) runChain(
```

golangci-lint (2.12.2)

[error] cyclomatic complexity 32 of func `(*investigationRunner).runChain` is high (> 30)

(gocyclo)